src/Device/Descriptor.cpp

```diff
@@ -483,7 +485,13 @@ try {
     msg.Format(_T("%s: %s (%s)"), _("Unable to open port"), name,
                (const TCHAR *)what);
 
-    env.SetErrorMessage(msg);
+    try {
+	    if (msg.length() > MAX_PATH-1)
+	      msg.Truncate(MAX_PATH-1);
+      env.SetErrorMessage(msg);
+    } catch (const std::exception &e) {
+      return false;
```

Comment:
> catch exception with wrong env variable

... why?

Reply:
I'll postpone this troubleshooting until later, for this PR it doesn't matter for the time being. I once got a buffer overflow here that I would have liked to have cleared - but I guess I'll have to do a loop again...

Reply:
What, a buffer overflow? Your commit messages does not say anything about a buffer overflow. And even if there were a buffer overflow - buffer overflows cannot be caught by a `catch` block. Nothing of this makes any sense.

Reply:
I have to do something more in this topic, why the formatted string is longer then allowed - but I will do it later, not yet - and I have postponed the commit.